engine/runtime/src/main/java/org/enso/interpreter/runtime/data/EnsoSourceSection.java

```diff
@@ -1,67 +1,81 @@
 package org.enso.interpreter.runtime.data;
 
+import com.oracle.truffle.api.CompilerDirectives.TruffleBoundary;
+import com.oracle.truffle.api.interop.InteropLibrary;
+import com.oracle.truffle.api.library.CachedLibrary;
+import com.oracle.truffle.api.library.ExportLibrary;
+import com.oracle.truffle.api.library.ExportMessage;
 import com.oracle.truffle.api.source.SourceSection;
+import org.enso.interpreter.runtime.EnsoContext;
+import org.enso.interpreter.runtime.data.text.Text;
+import org.enso.interpreter.runtime.data.vector.ArrayLikeHelpers;
 
 /**
  * Wrapper for exposing source sections in Enso. Delegates to the original methods with no behaviour
  * changes.
  */
-public class EnsoSourceSection {
+@ExportLibrary(InteropLibrary.class)
+public final class EnsoSourceSection implements EnsoObject {
+  private static final String[] MEMBERS = {
+    "getStartLine", //
+    "getEndLine", //
+    "getEndColumn", //
+    "getCharIndex", //
+    "getCharLength", //
+    "getCharEndIndex", //
+    "getCharacters", //
+    "getStartColumn", //
+    "isAvailable", //
+    "hasLines", //
+    "hasColumns", //
+    "hasCharIndex", //
+    "getSource", //
+  };
   private final SourceSection sourceSection;
 
   public EnsoSourceSection(SourceSection sourceSection) {
     this.sourceSection = sourceSection;
   }
 
-  public int getStartLine() {
-    return sourceSection.getStartLine();
+  @ExportMessage
+  boolean hasMembers() {
+    return true;
   }
 
-  public int getEndLine() {
-    return sourceSection.getEndLine();
+  @TruffleBoundary
+  @ExportMessage
+  Object readMember(String name, @CachedLibrary("this") InteropLibrary node) {
+    return switch (name) {
+      case "getStartLine" -> sourceSection.getStartLine();
```

Comment:
nitpick: Isn't there a more elegant way how to declare messages as enums so that this switch is checked to be complete?

Reply:
What is the overhead of `enum`? This code is behind `@TruffleBoundary`, but if it wasn't: Is `switch` on `enum` as effective as comparing `intern`ed strings?

> more elegant way

`enum` would be more elegant, but at the end we should have decent test coverage, right?

Reply:
Probably more important API question: is it OK that the identifiers are in _camel case_ and not _snake case_? I can report a bug for someone to fix it and ensure coverage is OK or the code is elegant.